File: problem_52.cc
#include<map>
#include<iostream>

unsigned int primed_product(int number)
{
	std::map< int, int> mappings;
	mappings[1] = 2;
	mappings[2] = 3;
	mappings[3] = 5;
	mappings[4] = 7;
	mappings[5] = 11;
	mappings[6] = 13;
	mappings[7] = 17;
	mappings[8] = 19;
	mappings[9] = 23;
	mappings[0] = 29;
   		
	unsigned int product = 1;
 
	while (number > 0)
	{
		int digits = number%10;
   		number /= 10;
		product *= mappings[digits];	
	}
	
	return product;
}

int main()
{
	bool found = false;
	unsigned int i = 1;
	
	while(found == false)
	{
		if(i % 10000 == 0)
		{
			std::cout << "\r" << std::flush;
			std::cout << i; 
		}
		unsigned int x = primed_product(i);
		if(primed_product(2*i) == x && primed_product(3*i) == x && primed_product(4*i) == x && primed_product(5*i) == x && primed_product(6*i) == x)
		{
			std::cout << "\r" << std::flush;
			std::cout << i << std::endl;
			found = true;
		}
		
		i++;
	}

	return 0;
}

File: problem_52.py
mappings = {1:2, 2:3, 3:5, 4:7, 5:11, 6:13, 7:17, 8:19, 9:23, 0:29}

def primed_list(x):
    return list(map(lambda y: mappings[int(y)], str(x)))

def primed_product(x):
    product = 1
    for i in primed_list(x):
        product *= i

    return product

found = False

i = 1
while found == False:
    x = primed_product(i)
    if primed_product(2*i) == x and primed_product(3*i) == x and primed_product(4*i) == x and primed_product(5*i) == x and primed_product(6*i) == x:
        print(i)
        found = True
    i +=1
